src/world/area_mgm/mgm_02/E15D80.c

```diff
@@ -1,39 +1,1003 @@
+    deciseconds = ((f32)(timeLeft % 30) * 10.0) / 30;
+    seconds = timeLeft / 30;
+
+    draw_box(0, 0xB, data->windowB_posX, 27, 0, 60, 20, 180, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, NULL, NULL, NULL, SCREEN_WIDTH, SCREEN_HEIGHT, NULL);
+    // draw whole seconds
+    draw_number(seconds, data->windowB_posX + 29, 31, TRUE, MSG_PAL_WHITE, 255, 3);
+    // draw tenths of seconds
+    draw_number(deciseconds, data->windowB_posX + 40, 31, TRUE, MSG_PAL_WHITE, 255, 0);
+    // draw dot
+    draw_msg(MESSAGE_ID(0x8,0x24), data->windowB_posX + 30, 29, 255, MSG_PAL_WHITE, 0);
+}
+
+void N(work_draw_score)(void) {
+    RenderTask task;
+
+    task.renderMode = RENDER_MODE_2D;
+    task.appendGfxArg = 0;
+    task.appendGfx = &mgm_02_draw_score_display;
+    task.distance = 0;
+
+    queue_render_task(&task);
+}
+
+ApiStatus N(CreateScoreDisplay)(Evt* script, s32 isInitialCall) {
+    SmashGameData* data = get_enemy(SCOREKEEPER_ENEMY_IDX)->varTablePtr[SMASH_DATA_VAR_IDX];
+    s32 hudElemA, hudElemMeter;
+
+    if (isInitialCall) {
+        data->workerID = create_generic_entity_world(NULL, &mgm_02_work_draw_score);
+
+        hudElemA = hud_element_create(HudScript_AButton);
+        data->hudElemID_AButton = hudElemA;
+        hud_element_set_render_depth(hudElemA, 0);
+        hud_element_set_flags(hudElemA, HUD_ELEMENT_FLAGS_80);
+        hud_element_set_tint(hudElemA, 255, 255, 255);
+        hud_element_set_script(hudElemA, HudScript_AButton);
+
+        hudElemMeter = hud_element_create(HudScript_BlueMeter);
+        data->hudElemID_Meter = hudElemMeter;
+        hud_element_set_render_depth(hudElemMeter, 0);
+        hud_element_set_flags(hudElemMeter, HUD_ELEMENT_FLAGS_80);
+    }
+
+    return ApiStatus_BLOCK;
+}
+
+ApiStatus N(DisableMenus)(Evt* script, s32 isInitialCall) {
+    gOverrideFlags |= GLOBAL_OVERRIDES_DISABLE_MENUS;
+    func_800E9894();
+    close_status_menu();
+    return ApiStatus_DONE2;
+}
+
+ApiStatus N(EnableMenus)(Evt* script, s32 isInitialCall) {
+    gOverrideFlags &= ~GLOBAL_OVERRIDES_DISABLE_MENUS;
+    return ApiStatus_DONE2;
+}
+
+ApiStatus N(DestroySignpost)(Evt* script, s32 isInitialCall) {
+    SmashGameData* data = get_enemy(SCOREKEEPER_ENEMY_IDX)->varTablePtr[SMASH_DATA_VAR_IDX];
+    delete_entity(data->signpostEntity);
+    return ApiStatus_DONE2;
+}
+
+ApiStatus N(CreateSignpost)(void) {
+    SmashGameData* data = get_enemy(SCOREKEEPER_ENEMY_IDX)->varTablePtr[SMASH_DATA_VAR_IDX];
+    s32 entityIndex = create_entity(&D_802EAFDC, 355, 20, -180, 0, 0, 0, 0, MAKE_ENTITY_END);
+    data->signpostEntity = entityIndex;
+    get_entity_by_index(entityIndex)->boundScriptBytecode = &D_80242A3C_E187BC;
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus N(OnHitBox)(Evt* script, s32 isInitialCall0) {
+    SmashGameData* data = get_enemy(SCOREKEEPER_ENEMY_IDX)->varTablePtr[SMASH_DATA_VAR_IDX];
+    s32 hitModelID;
+    s32 hitColliderID;
+    s32 i;
+
+    hitModelID = evt_get_variable(script, LW(0xA));
```

Comment:
I think we should move the initialization to the lines these are declared as well